superset/views/core.py

```diff
@@ -1330,11 +1335,32 @@ def testconn(self):
             with closing(engine.connect()) as conn:
                 conn.scalar(select([1]))
                 return json_success('"OK"')
-        except Exception as e:
-            logger.exception(e)
+        except NoSuchModuleError as e:
+            logger.info(f"Invalid driver {e}")
+            driver_name = make_url(uri).drivername
             return json_error_response(
-                "Connection failed!\n\n" f"The error message returned was:\n{e}", 400
+                _(
+                    "Could not load database driver: %(driver_name)s",
+                    driver_name=driver_name,
+                ),
+                400,
             )
+        except ArgumentError as e:
+            logger.info(f"Invalid URI {e}")
+            return json_error_response(
+                _(
+                    "Invalid connnection string, a valid string follows: \n"
+                    " 'DRIVER://USER:PASSWORD@DB-HOST/DATABASE-NAME'"
```

Comment:
The exact connection string varies from engine to engine. While adding a valid example to each engine spec might be overkill, it might be good to state that a valid connection string *usually* follows this pattern.

Reply:
nit: leading space in example connection string

Reply:
Done